Scripts/lua_singleton.gd

```diff
@@ -417,7 +418,17 @@ func _splitstr(input: String, separator: String):
 func _trim(input: String):
 	return input.strip_edges()
 
-func setup_extension(extension):
+func setup_extension(file):
+	if len(file.split(".")) == 1:
+		editor.warn("[color=yellow]WARNING[/color]: File has no extension.\nGriddyCode can't recognize it's language.\nFile will be parsed as raw text.")
+		return
+
+	var extension = file.split('.')[-1]
+
+	if extension in text_extensions:
+		editor.warn("[color=yellow]WARNING[/color]: File is text file, not code.")
+		return
```

Comment:
```suggestion
	if extension == "txt" :
		editor.warn("[color=yellow]WARNING[/color]: File is text file, not code.")
		return
```

There's literally only 1 extension in text_extensions. Why do we need an array right now?